### PR TITLE
RSS Block: Add border-box CSS

### PR DESCRIPTION
**Pull request by scruffian**
## Description
I think we should add `box-sizing: border-box` to the RSS block. The block has a fairly large amount of padding. If we set the block to be full width with `width: 100%` then we'll get horizontal scrollbars on the page because of the padding. This CSS will fix that.

## How has this been tested?
In TT1 Blocks

## Types of changes
Breaking change - this could potentially break existing implementations, although I think it is more likely to fix them.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by jasmussen** [comment]
This is a cool PR, but I'm seeing some weirdness that is probably unrelated to this PR, but also some that is. Screenshots after:

Screenshot 2020-12-17 at 08 53 41

Screenshot 2020-12-17 at 08 53 57

Frontend after:

Screenshot 2020-12-17 at 08 54 06

Before is the same, actually:

Screenshot 2020-12-17 at 08 54 59

It appears there's a ton of padding on the RSS feed block caused by this:

```
.wp-block-rss {
    padding-left: 2.5em;
}
```

that appears to affect both the placeholder state, and the rendered state, but only in the editor.

Screenshot 2020-12-17 at 08 55 32

Furthermore, in the editor the box-sizing property appears overridden by this:

Screenshot 2020-12-17 at 08 56 21

I appreciate trying to keep the PR small, and I think it would be fine to land this. But since you're here — are you seeing the same issues, and should we look to fix them at the same time?

**Comment by scruffian** [comment]
You're right, I removed the padding and added an `!important` to strengthen the border-box. There's a lot more we could do here, but I think it should happen as part of https://github.com/WordPress/gutenberg/issues/27766

**Comment by jasmussen** [comment]
This is cool:

Screenshot 2020-12-17 at 13 25 43

Screenshot 2020-12-17 at 13 25 49

But the need for an `!important` bugs me, and it seems to highlight an issue with the reset mixin having too high specificity. 

This seems to accomplish the same:

```
.wp-block-rss {

	// This needs extra specificity due to the reset mixin on the parent: https://github.com/WordPress/gutenberg/blob/a250e9e5fe00dd5195624f96a3d924e7078951c3/packages/edit-post/src/style.scss#L54
	&.wp-block-rss {
		box-sizing: border-box;
	}
```

👆 it will need a linter rule opt-out because the linter will say something about duplicate rules, I can't recall which rule. But it would elevate the specificity without going nuclear. 

What do you think?

**Comment by scruffian** [comment]
Yeah I thought about that option but I thought that generally people would be more familiar with the `!important`. I agree that it's not ideal, but I don't see the double selector as a much better alternative. I suppose it does have the benefit of being more easily modified by the theme.

 Another option could be to modify the mixin so it doesn't target the rss block - `*:not(.wp-block-rss)` but that also feels bad.

It feels like we need a way to gradually move towards setting border-box on everything, but we've quite aggressively prevented that in the past, so it's going to be a long road!

**Comment by jasmussen** [comment]
The problem with !important is that it's extremely hard to override, whereas the other approach adds just a little specificity, so I personally much prefer the other approach. 

But in order to not be blocked on this, I'm happy to defer to a sanity check from @kjellr or @MaggieCabrera.

File: packages/block-library/src/rss/style.scss
Comment (MaggieCabrera) [inline packages/block-library/src/rss/style.scss:4]: I prefer this much more than !important. I wonder why the theme css is not loaded after the editor's, wouldn't that solve all these kinds of issues?

Comment (kjellr) [reply packages/block-library/src/rss/style.scss:4]: Yeah, I prefer the double-selector approach here too. 👍

**Review by jasmussen — APPROVED** [review]
Let's get this in. Thanks Ben.

File: packages/block-library/src/rss/style.scss
Comment (youknowriad) [reply packages/block-library/src/rss/style.scss:4]: It seems the extra specificity is only needed in the frontend, so I wonder whether we should keep "style" pure and add the specificity in "editor.scss". Not a big deal though.

Comment (scruffian) [reply packages/block-library/src/rss/style.scss:4]: IMO the benefits of having the same code in the editor and frontend are greater so I'd prefer to keep it like this.

**Pull request merged**
